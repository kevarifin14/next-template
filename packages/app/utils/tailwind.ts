export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};
