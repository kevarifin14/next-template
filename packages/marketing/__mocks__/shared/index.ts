export * from 'shared';

export const mockPost = jest.fn();
export const post = mockPost;
