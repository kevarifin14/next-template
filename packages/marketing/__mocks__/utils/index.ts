import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
