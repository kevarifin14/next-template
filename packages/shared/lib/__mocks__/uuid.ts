export const mockUuid = '123456879';
export const v4 = jest.fn(() => mockUuid);
