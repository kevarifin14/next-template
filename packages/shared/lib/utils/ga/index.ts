export const pageview = (url) => {
  window.gtag("config", process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS, {
    page_path: url,
  });
};

type EventCategory =
  | "lead"
  | "outbound link"
  | "call to action"
  | "login"
  | "swap";

export const event = (
  category: EventCategory,
  action,
  label = "",
  params = {}
) => {
  if (process.env.NODE_ENV === "production") {
    window.gtag("event", action, {
      event_category: category,
      event_label: label,
      ...params,
    });
  }
};
