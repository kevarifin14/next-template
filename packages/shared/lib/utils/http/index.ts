import QueryString from 'query-string';

export function generateQueryString(params) {
  return QueryString.stringify(params);
}

export function generateEndpoint(endpoint, params) {
  if (Object.keys(params).length === 0) {
    return endpoint;
  }
  return `${endpoint}?${generateQueryString(params)}`;
}

export const get = async (endpoint, params = {}) => {
  const endpointWithParams = generateEndpoint(endpoint, params);
  const response = await fetch(endpointWithParams);

  if (response.status === 200) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};

export const post = async (endpoint, body, headers?) => {
  const response = await fetch(
    endpoint,
    {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'Content-Type': 'application/json',
        ...headers,
      },
    },
  );

  if (response.status >= 200 && response.status < 400) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};
