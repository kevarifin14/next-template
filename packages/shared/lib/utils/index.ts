export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
