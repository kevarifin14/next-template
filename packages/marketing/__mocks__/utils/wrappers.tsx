import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
