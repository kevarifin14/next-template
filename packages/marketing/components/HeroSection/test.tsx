import { render } from '@testing-library/react';

import { mockDescription, mockTitle } from '__mocks__/utils';
import { notificationsProviderWrapper } from '__mocks__/utils/wrappers';

import { HeroSection } from '.';

describe('HeroSection', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(
      <HeroSection title={mockTitle} description={mockDescription} withSubscribeForm />,
      { wrapper: notificationsProviderWrapper },
    );
    expect(baseElement).toMatchSnapshot();
  });
});
