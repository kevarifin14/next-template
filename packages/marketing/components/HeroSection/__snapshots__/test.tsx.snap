// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection matches snapshot 1`] = `
<body>
  <div>
    <div
      class="fixed max-w-sm w-full right-0 top-0 flex flex-col justify-end items-start space-y-4 pointer-events-none z-30 p-4"
    />
    <div
      class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-none"
    >
      <div
        class="max-w-7xl mx-auto text-center prose-xl dark:prose-light py-12"
      >
        <h1
          class="font-extrabold m-0"
        >
          <span
            class="mr-3"
          >
            mockTitle
          </span>
          <span
            class="text-primary"
          />
        </h1>
        <p
          class="mt-3"
        >
          mockDescription
        </p>
        <form
          class="flex space-x-4 max-w-lg mx-auto py-8"
        >
          <div
            class="w-full"
          >
            <div
              class="relative"
            >
              <input
                class="shadow-sm focus:outline-none w-full border bg-light border-light-400 rounded-md block text-gray-900 placeholder-gray-500 focus:border-primary-600 focus:ring-primary-600 focus:ring-1 dark:bg-dark-light dark:border-transparent dark:text-white px-4 py-2"
                id="123456879"
                name="email"
                placeholder="Enter Your Email"
                type="email"
              />
            </div>
          </div>
          <div
            class="flex-shrink-0"
          >
            <button
              class="inline-flex items-center justify-center focus:outline-none font-medium rounded-md text-white bg-gradient-to-tr from-primary-500 to-primary-800    px-4 py-2 "
              type="submit"
            >
              Get Early Access
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
